libdevice/spirv_vars.hpp

```diff
@@ -26,9 +26,9 @@ SYCL_EXTERNAL size_t __spirv_LocalInvocationId_z();
 #else // __SYCL_NVPTX__
 
 typedef size_t size_t_vec __attribute__((ext_vector_type(3)));
-extern "C" const __attribute__((opencl_constant))
+extern "C" const __attribute__((opencl_global))
```

Comment:
Can you clarify why this is changed?

Reply:
Once we re-use OpenCL address space attributes, we re-use diagnostics for them. It seems that OpenCL doesn't allow uninitialized variables in constant address space.

Reply:
Can we use default address space here?

Reply:
I think yes. Such change doesn't break my local `check-sycl` run on CPU device and I don't see any address space requirements in SPIR-V representation in LLVM IR [doc](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/docs/SPIRVRepresentationInLLVM.rst#spir-v-builtins-variables-mapped-to-llvm-global-variables) .

Reply:
I was troubled by this as well. I couldn't find a requirement from the OpenCL environment spec. 

@bashbaug shouldn't the env spec specify the storage class of builtin variables ?

Reply:
> shouldn't the env spec specify the storage class of builtin variables ?

Yes, I believe it should.  Would you mind filing an issue against the SPIR-V Environment spec on https://github.com/khronosgroup/opencl-docs to get this fixed?